packages/buidler-core/src/internal/buidler-evm/provider/provider.ts

```diff
@@ -300,15 +310,47 @@ export class BuidlerEVMProvider extends EventEmitter
     this._evmModule = new EvmModule(node);
     this._buidlerModule = new BuidlerModule(node);
 
-    const listener = (payload: { filterId: BN; result: any }) => {
-      this.emit("notifications", {
-        subscription: `0x${payload.filterId.toString(16)}`,
-        result: payload.result,
-      });
+    this._forwardNodeEvents(node);
+    this._emitConnectedEvent();
```

Comment:
I wonder if this should be emitted in the constructor instead?

Reply:
That's kind of a good point, cause it would be confusing emitting it here, as this is lazy.

But also, if emitted in the constructor, it's impossible to attach a listener before it's emitted.

Maybe we should just remove it.